xdsl/ir.py

```diff
@@ -54,8 +57,35 @@ class MLContext:
     registered_unregistered_ops: dict[str, type[Operation]] = field(
         default_factory=dict)
 
+    def _register_module(self, dialect: ModuleType):
```

Comment:
How about `dialect_from_module` here, that returns a Dialect? Would be easier to test also